admin/pycodestyle-twisted.py

```diff
@@ -0,0 +1,110 @@
+
+    E301: class Foo:\n    b = 0\n    def bar():\n        pass
+    E302: def a():\n    pass\n\ndef b(n):\n    pass
+    E302: def a():\n    pass\n\nasync def b(n):\n    pass
+    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
+    E303: def a():\n\n\n\n    pass
+    E304: @decorator\n\ndef a():\n    pass
+    E305: def a():\n    pass\na()
+    E306: def a():\n    def b():\n        pass\n    def c():\n        pass
+    """
+    top_level_lines = 3
+    method_lines = 2
+    if line_number < 3 and not previous_logical:
+        return  # Don't expect blank lines before the first line
+    if previous_logical.startswith('@'):
+        if blank_lines:
+            yield 0, "E304 blank lines found after function decorator"
+    elif (
+        (blank_lines > top_level_lines + 1) or
+        (indent_level and blank_lines == method_lines + 1)
+            ):
+        yield 0, "E303 too many blank lines (%d)" % blank_lines
+    elif pycodestyle.STARTSWITH_TOP_LEVEL_REGEX.match(logical_line):
+        if indent_level:
+            if not (blank_before or previous_indent_level < indent_level or
+                    pycodestyle.DOCSTRING_REGEX.match(previous_logical)):
+                ancestor_level = indent_level
+                nested = False
+                # Search backwards for a def ancestor or tree root (top level).
+                for line in lines[line_number - 2::-1]:
+                    if (
+                        line.strip() and
+                        pycodestyle.expand_indent(line) < ancestor_level
+                            ):
+                        ancestor_level = pycodestyle.expand_indent(line)
+                        nested = line.lstrip().startswith('def ')
+                        if nested or ancestor_level == 0:
+                            break
+                if nested:
+                    yield 0, "E306 expected 1 blank line before a " \
+                        "nested definition, found 0"
+                else:
+                    yield 0, "E301 expected 1 blank line, found 0"
+        elif blank_before != top_level_lines:
+            yield 0, "E302 expected %s blank lines, found %d" % (
+                top_level_lines, blank_before)
+    elif (
+        logical_line and
+        not indent_level and
+        blank_before != top_level_lines and
+        previous_unindented_logical_line.startswith(('def ', 'class '))
+            ):
+        yield 0, "E305 expected %s blank lines after " \
+            "class or function definition, found %d" % (
+                top_level_lines, blank_before)
+
+
+
+# Replace the upstream blank_lines implementation, with our fork.
+_previousValue = pycodestyle._checks['logical_line'][pycodestyle.blank_lines]
+del pycodestyle._checks['logical_line'][pycodestyle.blank_lines]
+pycodestyle._checks['logical_line'][blank_lines] = _previousValue
+
+
+
+if __name__ == '__main__':
+    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
```

Comment:
I don't understand what the purpose of this replacement is. Something to do with Windows? Can you explain?

Reply:
This is just copied from upstream... I think that this is how entrypoints scripts are installed .

see all the scripts from a virtualenv bin folder.